### DISCUSSION TITLE
What is preferred way to integrate libuv with MIDI IO processing?

### DISCUSSION THREAD

**Discussion by Oldes** [libuv/libuv#4519] [Q&A]
When I want to use libuv and also need to process MIDI, what is the best way to do it?
Using `uv_timer`, `uv_idle` or `uv_run` as `UV_RUN_NOWAIT`?

**Comment by bnoordhuis** [comment]
What does "process MIDI" mean exactly? Read from hardware ports, parse .mid files, something else?

**Reply by Oldes** [reply]
Read and writa data from MIDI hardware.

**Reply by bnoordhuis** [reply]
Are those blocking operations, i.e., can they stall for an indeterminate amount of time? If so, I'd probably spin up a separate thread and communicate with the event loop thread by means of uv_async_send() and a mutex-protected data structure (like a thread-safe queue.)
